### ISSUE TITLE
examinemem with address arithmetic

### ISSUE THREAD

**Issue by felixge** [go-delve/delve#2384]
This is a feature request or maybe this feature already exists and I didn't find it.

I'd like to be able to examine memory locations as expressed using address arithmetic. Example:

```
x 0x000000c000046778 - 8
```

Currently this gives me an error like this:

```
Command failed: convert address into uintptr type failed, strconv.ParseUint: parsing "0x000000c000046778-8": invalid syntax
```

It'd be nice if `examinemem` instead evaluated the arguments as an address arithmetic expression in order to lookup `0x000000c000046770`.

If this is interesting, I might be able to contribute the code for this.

**Comment by felixge** [comment]
This issue seems to be closely related to https://github.com/go-delve/delve/issues/2159. If `p` added support for hex output that would also help me. But perhaps this proposal would be simpler to implement.

**Comment by felixge** [comment]
The longer I used the delve command line, the more I want this 🙈. So I just hacked it into my local fork and it seems to work well. The patch is here: https://github.com/go-delve/delve/pull/2385

Please let me know what you think : )

**Comment by aarzilli** [comment]
Implemented by 7bf5482b328673e0b7dc3ebe7c6afe92c602aee2
